### PR TITLE
[plugin.video.catchuptvandmore] 0.1.10

### PR DESCRIPTION
**Pull request by SylvainCecchetto**
### Description

[Version 0.1.10]
[Add Live TV] DW (Contents EN, DE, ES, AR) (wwark)
[Fix] Default Quality for Replay group TF1 (Thanks Chris_wafer) (wwark)
[Readd Live TV] C8, CSTAR, Canalplus (clair) (wwark)
[Fix] Arte Replay and Add some new contents language (EN|ES|PL)
[Improvement] New menu layout and new live TV menu presentation (SylvainCecchetto)
[Upgrade] Upgrade SimplePlugin framework to v2.3.2 (SylvainCecchetto)
[Fix] Broken favorites (SylvainCecchetto)
[Fix] Live TV for RTL TVI, PlugRTL, ClubRTL (wwark)
[Add Channel] 日テレ (JP) (wwark)
[Add Channel] テレビ朝日 (JP) (wwark)
[Add Channel] TBSテレビ (JP) (wwark)
[Add Channel] テレビ東京 (JP) (wwark)
[Add Channel] MBSテレビ (JP) (wwark)
[Add Channel] 朝日放送株式会社 (JP) (wwark)
[Add Channel] 読売テレビ (JP) (wwark)
[Improvement] Use XMLTV Telerama for Live FR TV guide (SylvainCecchetto)
[Improvement] Faster navigation in plugin (SylvainCecchetto)
[Fix] RTBF Auvio - Replay Emissions (BE) (wwark)
[Improvement] Customizable parameters to download video (SylvainCecchetto)
[Fix] PublicSenat Replay (wwark)
[Fix] L'équipeTV Replay (wwark)
[Fix] RTC (BE) (wwark)

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.catchuptvandmore/resources/lib/openvpn.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
+# /*
+# *
+# * OpenVPN for Kodi.
```
Comment (Lunatixz) [inline plugin.video.catchuptvandmore/resources/lib/openvpn.py]: Inorder to merge this plugin you need to remove any code that circumvents geo restrictions. VPN's are not allowed.

**Comment by SylvainCecchetto** [comment]
Just removed everything concerning VPN ;-)

File: plugin.video.catchuptvandmore/resources/lib/root/channels/ca/tv5.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+import re
+from bs4 import BeautifulSoup as bs
+from resources.lib import utils
+from resources.lib import common
+
+# TO DO
+# Info Videos (date, plot, etc ...)
+
+
+URL_ROOT = 'https://%s.ca'
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/root/channels/ca/tv5.py]: not a big fan of this

File: plugin.video.catchuptvandmore/resources/lib/root/channels/fr/6play.py [hunk 3]
```diff
@@ -392,43 +351,71 @@ def list_videos(params):
...
-@common.PLUGIN.mem_cached(common.CACHE_TIME)
+#@common.PLUGIN.mem_cached(common.CACHE_TIME)
 def get_video_url(params):
     """Get video URL and start video player"""
     video_json = utils.get_webcontent(
         URL_JSON_VIDEO % (params.video_id),
         random_ua=True)
     json_parser = json.loads(video_json)
 
+    print 'COUCOU'
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/root/channels/fr/6play.py]: `xbmc.log` please. Coucou looks like a debug print that should go away :)

**Comment by SylvainCecchetto** [comment]
Thank you for your quick review. I just commit a modified version of the plugin.

I have one question, we have our own repo that contain only our plugin in order to offer people the ability to get beta with version like 0.1.11~beta5 and "official" releases with version like 0.1.11.

Do you think that it is possible to differentiate the version of the official Kodi repository and our personal repository by modifying respectively the version numbers in 0.1.10~kodirepo and 0.1.10~devrepo (for exemple). I don't know if Kodi can deals with that?

Thank you!

**Comment by enen92** [comment]
I personally don't see a problem with it. As long as your repo serves only for beta testing a specific addon and is not a general purpose repository that overrides versions for other plugins in the kodi official repository. You can label your beta versions as `x.x.x~betax` in your testing repo and the official/stable versions in our repo as `x.x.x`. Be aware that the availability of external repos is not something that will be always granted and might change in the future.

**Comment by SylvainCecchetto** [comment]
Yes I confirm that our repo (https://repo.cecchettosylvain.fr) contain only our plugin in two different versions:
- The last beta version labeled as x.y.z~betax
- The last stable version labeled as x.y.z

But in our repo the last stable version of the plugin contain the VPN interface feature (this feature only let the user to use the openvpn binary of the OS directly from the add-on. We don't provide any VPN connection or configuration to bypass geo restriction).

Our goal is to provide this version (with VPN interface) in our repo and the same version of the plugin (without VPN interface) on the official Kodi repo. This is why we want to find a correct way to deal with versions label of this two same versions with just the VPN interface as difference.

Thank you for your help.

We take note concerning the future evolution of Kodi concerning third repos ;)

**Comment by enen92** [comment]
I have to check with others but I think using our forum to distribute/advertise such repository is not allowed. We do not allow at all the discussion of techniques to bypass geo restrictions nor sharing of such code. VPNs are to be used at the OS level I don't understand why you even need to call the ovpn binary from kodi. If a user wants to use a VPN is free to do so, using it in the context of a plugin is plain wrong and only means the user is trying to bypass geolocation on a very specific context.

**Comment by SylvainCecchetto** [comment]
I understand your point of view. The only goal of this feature is to give user the opportunity to deal with openvpn binary from the plugin instead of SSH.

File: plugin.video.catchuptvandmore/resources/lib/root/channels/fr/live_tv_fr.py [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+
+
+def download_xmltv():
+    if os.path.exists(XMLTV_FILEPATH):
+        mtime = os.stat(XMLTV_FILEPATH).st_mtime
+        dl_file = (time.time() - mtime > 3600)
+    else:
+        dl_file = True
+    if dl_file:
+        r = requests.get("https://repo.cecchettosylvain.fr/xmltv/xmltv_fr.xml")
```
Comment (Lunatixz) [inline plugin.video.catchuptvandmore/resources/lib/root/channels/fr/live_tv_fr.py]: Where are you generating this XMLTV file from? XMLTV meta is licensed and usually only free to distribute under strict guidelines. ie if you are generating this via some type of unauthorized parsing script Kodi can not allow it. However if you are licensed to distribute this information or wrote a script to parse a websites meta to xmltv format you can include the parsing code in the script.

Comment (SylvainCecchetto) [reply plugin.video.catchuptvandmore/resources/lib/root/channels/fr/live_tv_fr.py]: I generate this XMLTV file on my personal server with a script that scrape data from a free French TV guide website.
Is it ok?

Comment (enen92) [reply plugin.video.catchuptvandmore/resources/lib/root/channels/fr/live_tv_fr.py]: I assumed this was the case...also noticed this while reviewing. We have a takedown policy anyway so even if this is violating some company terms of use they can always submit a complain.

**Comment by enen92** [comment]
I'd like to merge this today if there are no objections

**Pull request merged**